seed/java-sdk/bytes/src/main/java/com/seed/bytes/resources/service/ServiceClient.java

```diff
@@ -58,4 +61,9 @@ public void upload(byte[] request, RequestOptions requestOptions) {
             throw new SeedBytesException("Network error executing HTTP request", e);
         }
     }
+
+    // Overload for backward compatibility
+    public void upload(byte[] request, RequestOptions requestOptions) {
```

Comment:
thoughts on moving the direct bytes request to this?

Claude gave me these downsides but to me seems worth the cleanliness

> Potential downsides of the upload overload:

Memory Usage:

The original implementation used the byte array directly.
The new overload wraps the byte array in a ByteArrayInputStream, which is then wrapped in a FileStream.
While ByteArrayInputStream is generally efficient, it does create an additional object in memory.


Performance Overhead:

There's a slight performance cost in creating the additional objects (ByteArrayInputStream and FileStream).
The streaming approach might introduce a small amount of overhead compared to sending the byte array directly, especially for small files.

Reply:
great idea